open_bus_gtfs_etl/gtfs_loader.py

```diff
@@ -57,18 +58,57 @@ def _get_rides(self) -> List[Ride]:
                        self.all_trip_id.split(";"),
                        self.all_trip_id_to_date.split(";"))]
 
-    def _get_stops(self) -> List[RouteStop]:
+    def _upsert_stops(self, stops: Dict[int, Stop], stop_code, stop_latlon, stop_name, stop_desc_city) -> Stop:
+        """
+        Upsert (Update or Insert) Stops. Based on given dict of stops that represent the current up to date stops,
+        this method validate that new Stop will be created just in case its not already exist (with the same details)
+        Args:
+            stops: a dict the represent the current up to date stops
+            stop_code: stop code that identify the stop
+            stop_latlon: lat & lon of a stop
+            stop_name: name of the stop
+            stop_desc_city: stop's city
+
+        Returns: Stop
+        """
+        lat, lon = stop_latlon.split(",")
+
+        stop = stops.get(int(stop_code))
+
+        if stop is None or stop.lat != float(lat) or stop.lon != float(lon) \
+                or stop.name != stop_name or stop.city != stop_desc_city:
```

Comment:
@AvivSela please reply

Reply:
@OriHoch, yes. I'm fetching the prior or valid stop record for each stop and validate that new one will be created just in case stop is not exist, or it is different.

there are many cases i thought about. you can see them here: 
[def _upsert_stop(stops: Dict[int, Stop], stop_to_upsert: StopModel) -> Stop:](https://github.com/hasadna/open-bus-gtfs-etl/blob/16c313cd416a8d4a0ae8d9bbec035c862f51f7a4/open_bus_gtfs_etl/gtfs_loader.py#L105)